lib/model/rwfolder.go

```diff
@@ -1113,6 +1113,18 @@ func (p *rwFolder) copierRoutine(in <-chan copyBlocksState, pullChan chan<- pull
 		p.model.fmut.RUnlock()
 
 		for _, block := range state.blocks {
+			if state.reused == 0 && block.IsEmpty() {
```

Comment:
Because if we're reusing, the contents of the reused file is obviously not all-zeroes here (or we would not have this block to pull), so we must overwrite. We could potentially do another shortcut here, where we just write a block of zeroes instead of pulling it all the way over the network, but I think this will be a fairly rare branch, essentially only happening when we're reusing and don't have a block of zeroes available anywhere on disk.